File: mobile/lib/models/risk.dart
class AreaRisk {
  final double latitude;
  final double longitude;
  final double riskScore;
  final String riskCategory;
  final double confidence;
  final String summary;
  final List<String> contributingFactors;
  final int nearbyIncidentsCount;
  final double? nearestPoliceKm;
  final double? nearestHospitalKm;
  final String modelVersion;

  AreaRisk({
    required this.latitude,
    required this.longitude,
    required this.riskScore,
    required this.riskCategory,
    required this.confidence,
    required this.summary,
    required this.contributingFactors,
    required this.nearbyIncidentsCount,
    this.nearestPoliceKm,
    this.nearestHospitalKm,
    required this.modelVersion,
  });

  factory AreaRisk.fromJson(Map<String, dynamic> json) {
    return AreaRisk(
      latitude: (json['latitude'] as num).toDouble(),
      longitude: (json['longitude'] as num).toDouble(),
      riskScore: (json['risk_score'] as num).toDouble(),
      riskCategory: json['risk_category'] as String,
      confidence: (json['confidence'] as num).toDouble(),
      summary: json['summary'] as String,
      contributingFactors: (json['contributing_factors'] as List<dynamic>?)?.map((e) => e.toString()).toList() ?? [],
      nearbyIncidentsCount: json['nearby_incidents_count'] as int? ?? 0,
      nearestPoliceKm: (json['nearest_police_km'] as num?)?.toDouble(),
      nearestHospitalKm: (json['nearest_hospital_km'] as num?)?.toDouble(),
      modelVersion: json['model_version'] as String? ?? 'sakhi-risk-xgb-v1.0',
    );
  }
}

File: backend/app/schemas/risk.py
from typing import List, Optional, Dict
from pydantic import BaseModel, Field


class AreaRiskRequest(BaseModel):
    latitude: float = Field(..., ge=-90.0, le=90.0)
    longitude: float = Field(..., ge=-180.0, le=180.0)
    radius_km: float = Field(1.0, ge=0.1, le=10.0)


class ContributingFactor(BaseModel):
    name: str
    impact: str  # e.g., "Increased reported night incidents in last 30 days" or "Nearby emergency station within 400m"


class AreaRiskResponse(BaseModel):
    latitude: float
    longitude: float
    risk_score: float = Field(..., ge=0.0, le=1.0)
    risk_category: str  # lower_reported_risk, moderate_reported_risk, higher_reported_risk
    confidence: float
    summary: str
    contributing_factors: List[str]
    nearby_incidents_count: int
    nearest_police_km: Optional[float] = None
    nearest_hospital_km: Optional[float] = None
    model_version: str


class RoutePoint(BaseModel):
    latitude: float
    longitude: float


class RouteOption(BaseModel):
    name: str
    distance_km: float
    estimated_minutes: int
    risk_category: str
    risk_score: float
    explanation: str


class RouteRiskRequest(BaseModel):
    origin_lat: float
    origin_lon: float
    destination_lat: float
    destination_lon: float


class RouteRiskResponse(BaseModel):
    routes: List[RouteOption]
